File: leetcode/2195-time-needed-to-buy-tickets/solution.go
func timeRequiredToBuy(tickets []int, k int) int {
    ret := 0
    for i, x := range tickets {
        if i > k {
            ret += min(x, tickets[k]-1)
        } else {
            ret += min(x, tickets[k])
        }
    }
    return ret
}

File: leetcode/2195-time-needed-to-buy-tickets/solution.rs
impl Solution {
    pub fn time_required_to_buy(tickets: Vec<i32>, k: i32) -> i32 {
        use std::cmp::min;
        let k = k as usize;
        let cap = tickets[k];
        tickets.into_iter()
            .enumerate()
            .map(|(i,x)| if i > k { min(x, cap-1) } else { min(x, cap) })
            .sum()
    }
}
